File: civitas/src/civitas/SorpresaEspeculador.java
package civitas;


import java.util.ArrayList;

/**
 *
 * @author Usuario
 */
public class SorpresaEspeculador extends Sorpresa {
    
    private int fianza;
    
    SorpresaEspeculador(int fianza, String texto) {
        super(texto);
        this.fianza = fianza;
    }
    
    @Override
    public void aplicarAJugador(int actual, ArrayList<Jugador> todos) {
        if(jugadorCorrecto(actual, todos)) {
            informe(actual, todos);
            JugadorEspeculador jugador = new JugadorEspeculador(
                    todos.get(actual), fianza);
            todos.set(actual, jugador);
        }
    }
    
    @Override
    public String toString() {
        String salida = "Fianza: " + fianza + super.toString();
        return salida;
    }
    
}

File: civitasRuby/pkg/civitas-0.0.1/lib/sorpresa_especulador.rb
# encoding:utf-8
module Civitas

  class SorpresaEspeculador < Sorpresa
  
    def initialize(texto, fianza)
      super(texto)
      @fianza = fianza
    end

    def aplicar_a_jugador(actual, todos)
      if jugador_correcto(actual, todos)
        super(actual, todos)
        jugador = JugadorEspeculador.new(todos.at(actual), @fianza)
        todos[actual] = jugador
      end
    end
    
    def to_string
      return super.to_string + "\nFianza: #{@fianza}" 
    end
    
  public_class_method :new  
  end
end